hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/ethereum/Eip1559EthereumTransactionParser.java

```diff
@@ -0,0 +1,76 @@
+package com.hedera.mirror.importer.parser.record.ethereum;
+
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.esaulpaugh.headlong.rlp.RLPDecoder;
+import javax.inject.Named;
+
+import com.hedera.mirror.common.domain.transaction.EthereumTransaction;
+import com.hedera.mirror.importer.exception.InvalidDatasetException;
+
+@Named
+public class Eip1559EthereumTransactionParser implements EthereumTransactionParser {
+    private static final int EIP1559_TYPE_BYTE = 2;
+    private static final int EIP1559_TYPE_RLP_ITEM_COUNT = 12;
+
+    private static final String DECODE_ERROR_PREFIX_MESSAGE = "Unable to decode EIP1559 ethereum transaction bytes,";
+
+    @Override
+    public EthereumTransaction decode(byte[] transactionBytes) {
+        var decoder = RLPDecoder.RLP_STRICT.sequenceIterator(
+                transactionBytes);
+        var legacyRlpItem = decoder.next();
+        var legacyRlpItemByte = legacyRlpItem.asByte();
+        if (legacyRlpItemByte != EIP1559_TYPE_BYTE) {
+            throw new InvalidDatasetException(String.format("%s 1st byte was %s but should be %s",
+                    DECODE_ERROR_PREFIX_MESSAGE, legacyRlpItemByte, EIP1559_TYPE_BYTE));
+        }
+
+        var eip1559RlpItem = decoder.next();
+        if (!eip1559RlpItem.isList()) {
+            throw new InvalidDatasetException(String.format("%s 2nd RLPItem was not a list",
+                    DECODE_ERROR_PREFIX_MESSAGE));
+        }
+
+        var rlpItems = eip1559RlpItem.asRLPList().elements();
+        if (rlpItems.size() != EIP1559_TYPE_RLP_ITEM_COUNT) {
+            throw new InvalidDatasetException(String.format("%s 2nd RLPItem list size was %s but should be %s",
+                    DECODE_ERROR_PREFIX_MESSAGE, rlpItems.size(), EIP1559_TYPE_RLP_ITEM_COUNT));
```

Comment:
Create a custom exception that extends `InvalidDatasetException ` but simplifies this to:
```suggestion
            throw new EthereumTransactionException("2nd RLPItem list size was %s but should be %s", rlpItems.size(), EIP1559_TYPE_RLP_ITEM_COUNT));
```

Reply:
Adding the following so it's more general and can be used by multiple parsers for multiple scenarios
```java
public class InvalidEthereumBytesException extends InvalidDatasetException {

    ...

    private static final String DECODE_ERROR_PREFIX_MESSAGE = "Unable to decode %s ethereum transaction bytes, %s";

    public InvalidEthereumBytesException(String ethereumTransactionType, String message) {
        super(String.format(DECODE_ERROR_PREFIX_MESSAGE, ethereumTransactionType, message));
    }
}
```

Reply:
Added custom exception and updated references